### PR TITLE
Added missing workshop incremental buildings to mission requirements

### PR DESCRIPTION
**Pull request by Joaqim**
A lot of missions could only be fulfilled with X number of:
 * Workshop buildings 
or 
 * Counting Houses
completely ignoring the intermittent buildings:
 * Grand Workshop
 * Huge Workshop

### CODE REVIEW

File: missions/ME_000_Westphalian_Minors_Missions.txt [hunk 1]
```diff
@@ -284,6 +284,9 @@ westphalian_minor_missions_missions_4 = {
 		position = 1
 		trigger = {
 			workshop = 3
```
Comment (Grociu) [inline missions/ME_000_Westphalian_Minors_Missions.txt]: `
OR = {
    workshop = 3
    production_lvl_2 = 3
    counting_house = 3
    production_lvl_3 = 3
}
`

Comment (Grociu) [reply missions/ME_000_Westphalian_Minors_Missions.txt]: While still not perfect, this is what you want in here I think.

File: missions/ME_000_Westphalian_Minors_Missions.txt [hunk 2]
```diff
@@ -284,6 +284,9 @@ westphalian_minor_missions_missions_4 = {
...
+ 			production_lvl_2 = 3
+			counting_house = 3
+ 			production_lvl_3 3
 			temple = 3
```
Comment (Grociu) [inline missions/ME_000_Westphalian_Minors_Missions.txt]: OR = { temple = 3 taxation_lvl_2 =3  cathedral = 3 taxation_lvl_4 =3}

**Review by Grociu — CHANGES_REQUESTED** [review]
Good catch! Wesphalian minors need small adjustment, but rest is OK!

**Comment by Joaqim** [comment]
Noticed there were some other 
OR = {
       temple = 5
       cathedral = 5
}
instead of
OR = {
     temple = 5
     taxation_lvl_2 = 5
     cathedral = 5
     taxation_lvl_4 = 5
}

Should that be changed as well? I'll look into it more.

**Comment by Grociu** [comment]
> 
> 
> Noticed there were some other
> OR = {
> temple = 5
> cathedral = 5
> }
> instead of
> OR = {
> temple = 5
> taxation_lvl_2 = 5
> cathedral = 5
> taxation_lvl_4 = 5
> }
> 
> Should that be changed as well? I'll look into it more.

Yes if we're following the same logic you did with the production buildings. Probably would depend on case by case study of the missions, but in general it's better.

File: missions/ME_Caspian_Minor_Missions.txt [hunk 3]
```diff
@@ -374,6 +374,9 @@ cpm_missions_5 = {
 		position = 3
 		trigger = {
 			temple = 3
```
Comment (Grociu) [inline missions/ME_Caspian_Minor_Missions.txt]: Probably should be OR ={} for the buildings.

File: missions/ME_Irish_Minors_Missions.txt [hunk 4]
```diff
@@ -499,11 +501,12 @@ ME_irish_minors_missions_5 = {
 		position = 6
 		trigger = {
 			cathedral = 3
```
Comment (Grociu) [inline missions/ME_Irish_Minors_Missions.txt]: Or = {} ?

File: missions/ME_Spain_Missions.txt [hunk 5]
```diff
@@ -2589,6 +2591,7 @@ SPA_missions_4 = {
 		required_missions = { spain_build_churches }
 		trigger = {
 			cathedral = 10
```
Comment (Grociu) [inline missions/ME_Spain_Missions.txt]: OR = {}

**Comment by Joaqim** [comment]
My branch has become a little bit messy; trying to rebase to make it less verbose, but failing miserably. I have gone through all cases where there are temple requirements.

Spain has a notable exception in spain_build_cathedrals where the requirement is 'cathedral' or the next level 'taxation_lvl_4'

And there are some missions titled "early_game_buildings where I have for now included all the incremental buildings as triggers, just so the player doesn't get locked out from a mission tree for building late into the game.

File: missions/ME_Spain_Missions.txt (hunk 5 above)
Comment (Joaqim) [reply missions/ME_Spain_Missions.txt]: Fixed ME_Spain_Missions.txt

**Comment by Grociu** [comment]
Check this comment to avoid git issues in the future: https://github.com/TheDivexz/excelmakesbelhappypack/pull/72#issuecomment-627782908

**Comment by Joaqim** [comment]
I have many more missing OR ={}, I'll solve it in batch. Easy to find with grep

EDIT: Nvm, got a lot of false positives.

File: missions/ME_Caspian_Minor_Missions.txt [hunk 6]
```diff
@@ -373,8 +373,13 @@ cpm_missions_5 = {
...
 		position = 3
 		trigger = {
-			temple = 3
-			religious_unity = 1
+			OR = {
+				temple = 3
+ 				taxation_lvl_2 = 3
+				cathedral = 3
+ 				taxation_lvl_4 = 3
+				religious_unity = 1
```
Comment (Grociu) [inline missions/ME_Caspian_Minor_Missions.txt]: religious_unity = 1 should be outside the OR

**Review by Grociu — CHANGES_REQUESTED** [review]
One small issue, but probably the last :)

File: missions/ME_Caspian_Minor_Missions.txt (hunk 6 above)
Comment (Joaqim) [reply missions/ME_Caspian_Minor_Missions.txt]: Thanks, good catch.

**Review by Grociu — APPROVED** [review]

**Comment by Joaqim** [comment]
Sorry again for the mess. I just bach scripts to automate a lot. I'll make a cleaner branch next time I work on something for this project. 

Thanks for the help

**Comment by Grociu** [comment]
Thank you for your contribution! See you soon!

**Comment by stackpoint** [comment]
For something like this, using a scripted trigger may be useful. I believe ME actually has one that I updated with the IV buildings: PROVINCE_HAS_BUILDING_OF_GROUP in missions_expanded_buildings_scripted_triggers.txt

**Comment by stackpoint** [comment]
For the missions that count the buildings specifically, something like this:
```
calc_true_if = {
	all_owned_provinces = {
		 OR = {
		 	has_building = temple
		 	has_building = taxation_lvl_2
		 	has_building = cathedral
		 	has_building = taxation_lvl_4
		 }
	}
	amount = 5
}
```

**Comment by Grociu** [comment]
Would you be able to use the scripted trigger you mentioned to count this somehow?

**Comment by stackpoint** [comment]
You can't use a scripted trigger within a scripted trigger. So it depends on how robust calc_true_if is. So maybe.

File: missions/ME_000_Timurid_Princes_Missions.txt [hunk 7]
```diff
@@ -377,7 +377,9 @@ Timurid_princes_missions_5 = {
 		trigger = {
 			OR = {
 				temple = 10
+ 				taxation_lvl_2 = 10
 				cathedral = 10
+ 				taxation_lvl_4 = 10
 			}
 			calc_true_if = {
 				all_province = {
```
Comment (stackpoint) [inline missions/ME_000_Timurid_Princes_Missions.txt:389]: I know this isn't your fault but this scope needs to be changed.

**Comment by Grociu** [comment]
I did some testing and `PROVINCE_HAS_BUILDING_OF_GROUP` family of scripted triggers can be used here:
Syntax for this would be:
PRODUCTION (counts whether you have 10 or more prod buildings):
```
trigger = {
	calc_true_if = {
		all_owned_province = {
			custom_trigger_tooltip = {
				tooltip = PROVINCE_HAS_BUILDING_OF_GROUP_PRODUCTION_ALL
				province_has_building_of_group = { group = production all = yes }
			}
		}
		amount = 10
	}
}
```
TAXATION:
```
trigger = {
	calc_true_if = {
		all_owned_province = {
			custom_trigger_tooltip = {
				tooltip = PROVINCE_HAS_BUILDING_OF_GROUP_TAXATION_ALL
				province_has_building_of_group = { group = taxation all = yes }
			}
		}
		amount = 10
	}
}
```
For that one Castille mission, use what Stackpoint suggested. Conditional PROVINCE_HAS_BUILDING_OF_GROUP do exist, but updating them to fit that case, would take a lot of work.
These currently display an incorrect tooltip in game, but I'm going to fix that today. They work as intended.

**Comment by Grociu** [comment]
In spots where there was more than 1 building or the same type, I used the province_has_building_of_group triggers. In several places they were modified to for example check for 'buildings better than cathedral'
Used copy and paste a lot, so if there is a mistake in one, it's the same for all.
I tried to be conscious to change the numbers to what was in the original, but might have missed something in this department.

File: missions/ME_Irish_Minors_Missions.txt [hunk 8]
```diff
@@ -353,9 +353,14 @@ ME_irish_minors_missions_4 = {
...
-				workshop = 3
-				counting_house = 3
+			calc_true_if = {
+				all_owned_province = {
+					custom_trigger_tooltip = {
+						tooltip = PROVINCE_HAS_BUILDING_OF_GROUP_TAXATION_ALL
+						province_has_building_of_group = { group = taxation all = yes }
+					}
+				}
+				amount = 3
```
Comment (stackpoint) [inline missions/ME_Irish_Minors_Missions.txt:363]: production buildings?

File: missions/ME_Irish_Minors_Missions.txt [hunk 9]
```diff
@@ -463,7 +468,9 @@ ME_irish_minors_missions_5 = {
 		trigger = {
 			OR = {
 				temple = 3
+ 				taxation_lvl_2 = 3
 				cathedral = 3
+ 				taxation_lvl_4 = 3
```
Comment (stackpoint) [inline missions/ME_Irish_Minors_Missions.txt]: province_has_building_of_group here?

File: missions/ME_Maya_Missions.txt [hunk 10]
```diff
@@ -2595,10 +2607,24 @@ maya_missions_5 = {
...
+				amount = 30
+			}
+			calc_true_if = {
+				all_owned_province = {
+					custom_trigger_tooltip = {
+						tooltip = PROVINCE_HAS_BUILDING_OF_GROUP_TAXATION_EOMAT_cathedral
+						province_has_building_of_group = { group = taxation equal_or_more_advanced_than = cathedral }
+					}
+				}
+				amount = 10
```
Comment (stackpoint) [inline missions/ME_Maya_Missions.txt]: These need to be in an OR?

**Review by stackpoint — CHANGES_REQUESTED** [review]
Good job so far, looks like a big mess to start with.

File: missions/ME_Maya_Missions.txt (hunk 10 above)
Comment (Grociu) [reply missions/ME_Maya_Missions.txt]: Oh! I misinterpreted the original. Thought it required 30 temples and 10 cathedrals.

**Review by stackpoint — APPROVED** [review]
Looks good to go?

**Comment by mellester** [comment]
Look good to me too, when this gets added use the squash and merge button

**Comment by stackpoint** [comment]
```

**Comment by stackpoint** [comment]
Insert pun here: 
```
[persistent.cpp:52]: Error: "Unexpected token: calc_true_if, near line: 388
" in file: "missions/ME_000_Timurid_Princes_Missions.txt" near line: 393
[persistent.cpp:52]: Error: "Unexpected token: add_country_modifier, near line: 396
" in file: "missions/ME_000_Timurid_Princes_Missions.txt" near line: 399
[trigger.cpp:531]: Unknown trigger-type trigger in missions/ME_Irish_Minors_Missions.txt line : 509
[persistent.cpp:52]: Error: "Unexpected token: trigger, near line: 509
" in file: "missions/ME_Irish_Minors_Missions.txt" near line: 519
```

**Pull request merged**